### PR TITLE
Filling in layout

### PR DESCRIPTION
**Pull request by phucynwa**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -6,6 +6,8 @@ ruby '2.5.1'
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
```
Comment (mdmahmudurrahman) [inline Gemfile]: Please remove all of the comments.

File: Gemfile [hunk 2]
```diff
@@ -6,6 +6,8 @@ ruby '2.5.1'
...
 gem 'rails', '~> 5.2.0'
```
Comment (mdmahmudurrahman) [inline Gemfile]: Instead of using single quote, please use double quote.

File: app/assets/javascripts/static_pages.coffee [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (mdmahmudurrahman) [inline app/assets/javascripts/static_pages.coffee]: Remove this unused file.

File: app/assets/javascripts/users.coffee [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (mdmahmudurrahman) [inline app/assets/javascripts/users.coffee]: Remove this unused file.

File: app/assets/stylesheets/custom.scss [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  font-weight: normal;
+  color: $gray-light;
+}
+
+p {
+  font-size: 1.1em;
+  line-height: 1.7em;
+}
+
+
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Remove this extra new line

File: app/assets/stylesheets/custom.scss [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  }
+  ul {
+    float: right;
+    list-style: none;
+    li {
+      float: left;
+      margin-left: 15px;
+    }
+  }
+}
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Add a line break here.

File: app/assets/stylesheets/static_pages.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPages controller here.
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/static_pages.scss]: Remove this unused file.

File: app/assets/stylesheets/users.scss [hunk 8]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/users.scss]: Remove this unused file.

File: app/controllers/static_pages_controller.rb [hunk 9]
```diff
@@ -0,0 +1,13 @@
+class StaticPagesController < ApplicationController
+	def home
```
Comment (mdmahmudurrahman) [inline app/controllers/static_pages_controller.rb]: Use only 2 space indentation and correct it for the whole file.

File: app/helpers/application_helper.rb [hunk 10]
```diff
@@ -1,2 +1,10 @@
 module ApplicationHelper
+	def full_title(page_title = '')
+		base_title = "Ruby on Rails Tutorial Sample App"
+		if page_title.empty?
+			base_title
+		else
+			page_title + " | " + base_title
+		end
+	end
```
Comment (mdmahmudurrahman) [inline app/helpers/application_helper.rb]: Use only 2 space indentation and correct it for the whole file.

File: app/helpers/static_pages_helper.rb [hunk 11]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
```
Comment (mdmahmudurrahman) [inline app/helpers/static_pages_helper.rb]: Remove this unused file.

File: app/helpers/users_helper.rb [hunk 12]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
```
Comment (mdmahmudurrahman) [inline app/helpers/users_helper.rb]: Remove this unused file.

File: app/views/layouts/_footer.html.erb [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+    by <a href="http://www.michaelhartl.com/">Michael Hartl</a>
+  </small>
+  <nav>
+    <ul>
+      <li><%= link_to "About",   about_path %></li>
+      <li><%= link_to "Contact", contact_path %></li>
+      <li><a href="http://news.railstutorial.org/">News</a></li>
+    </ul>
+  </nav>
+</footer>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_footer.html.erb]: Add a line break here.

File: app/views/layouts/_footer.html.erb [hunk 14]
```diff
@@ -0,0 +1,13 @@
+<footer class="footer">
+  <small>
+    The <a href="http://www.railstutorial.org/">Ruby on Rails Tutorial</a>
+    by <a href="http://www.michaelhartl.com/">Michael Hartl</a>
+  </small>
+  <nav>
+    <ul>
+      <li><%= link_to "About",   about_path %></li>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_footer.html.erb]: Use only 1 space between the parameters.

File: app/views/layouts/_header.html.erb [hunk 15]
```diff
@@ -0,0 +1,12 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+    <%= link_to "sample app", root_path, id: "logo" %>
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to "Home",   root_path %></li>
+        <li><%= link_to "Help",   help_path %></li>
+        <li><%= link_to "Log in", '#' %></li>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_header.html.erb]: Use double quote

File: app/views/layouts/_header.html.erb [hunk 16]
```diff
@@ -0,0 +1,12 @@
...
+      </ul>
+    </nav>
+  </div>
+</header>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_header.html.erb]: Add a line break.

File: app/views/layouts/_header.html.erb [hunk 17]
```diff
@@ -0,0 +1,12 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+    <%= link_to "sample app", root_path, id: "logo" %>
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to "Home",   root_path %></li>
+        <li><%= link_to "Help",   help_path %></li>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_header.html.erb]: Use only 1 space between the parameters.

File: app/views/layouts/_shim.html.erb [hunk 18]
```diff
@@ -0,0 +1,4 @@
+<!--[if lt IE 9]>
+  <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/r29/html5.min.js">
+  </script>
+<![endif]-->
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_shim.html.erb]: Add a line break.

File: app/views/layouts/_rails_default.html.erb [hunk 19]
```diff
@@ -0,0 +1,4 @@
+<%= csrf_meta_tags %>
+  <%= stylesheet_link_tag    'application', media: 'all',
+  'data-turbolinks-track': 'reload' %>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_rails_default.html.erb]: When you break a line, add 2 space indentation to it

File: app/views/layouts/application.html.erb [hunk 20]
```diff
@@ -1,15 +1,16 @@
...
+  <%= render 'layouts/shim' %>
 
-    <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
-    <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
-  </head>
-
-  <body>
+</head>
+<body>
+  <%= render 'layouts/header' %>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/application.html.erb]: Use double quote.

File: app/views/layouts/application.html.erb [hunk 21]
```diff
@@ -1,15 +1,16 @@
...
+  <div class="container">
     <%= yield %>
-  </body>
-</html>
+    <%= render 'layouts/footer' %>
+  </div>
+</body>
+</html>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/application.html.erb]: Add a line break at the end of the file.

File: app/views/static_pages/about.html.erb [hunk 22]
```diff
@@ -0,0 +1,11 @@
...
+<h1>About</h1>
+<p>
+	The <a href="http://www.railstutorial.org/"><em>Ruby on Rails
+	Tutorial</em></a> is a
+	<a href="http://www.railstutorial.org/book">book</a> and
+	<a href="http://screencasts.railstutorial.org/">screencast series</a>
+	to teach web development with
+	<a href="http://rubyonrails.org/">Ruby on Rails</a>.
+	This is the sample application for the tutorial.
+</p>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/about.html.erb]: Please add a line break at the end of the file and use on 2 space indentation for the previous lines.

File: app/views/static_pages/contact.html.erb [hunk 23]
```diff
@@ -0,0 +1,6 @@
+<% provide(:title, "Contact") %>
+<h1>Contact</h1>
+<p>
+	Contact the Ruby on Rails Tutorial about the sample app at the
+	<a href="http://www.railstutorial.org/contact">contact page</a>.
+</p>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/contact.html.erb]: Same for this file like as the previous comment.

File: app/views/static_pages/help.html.erb [hunk 24]
```diff
@@ -0,0 +1,9 @@
+<% provide(:title, "Help") %>
+<h1>Help</h1>
+<p>
+	Get help on the Ruby on Rails Tutorial at the
+	<a href="http://www.railstutorial.org/help">Rails Tutorial help section</a>.
+	To get help on this sample app, see the
+	<a href="http://www.railstutorial.org/book"><em>Ruby on Rails Tutorial</em>
+	book</a>.
+</p>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/help.html.erb]: Same comment for this file like as the previous comment.

File: app/views/static_pages/home.html.erb [hunk 25]
```diff
@@ -0,0 +1,17 @@
...
+	</h2>
+
+	<%= link_to "Sign up now!", signup_path, class: "btn btn-lg btn-primary" %>
+
+</div>
+
+<%= link_to image_tag("rails.png", alt: "Rails logo"),
+'http://rubyonrails.org/' %>
+<%#= link_to image_tag("kitten.jpg", alt: "Kitten image"),
+            'http://rubyonrails.org/' %>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/home.html.erb]: Same comment for this file like as the previous comment and remove the comments.

File: app/views/users/new.html.erb [hunk 26]
```diff
@@ -0,0 +1,3 @@
+<% provide(:title, 'Sign up') %>
+<h1>Sign up</h1>
+<p>This will be a signup page for new users.</p>
```
Comment (mdmahmudurrahman) [inline app/views/users/new.html.erb]: Add a line break at the end of the file.

File: config/routes.rb [hunk 27]
```diff
@@ -1,3 +1,9 @@
 Rails.application.routes.draw do
+  get 'users/new'
+	root 'static_pages#home'
+	get  '/help', 	to: 'static_pages#help'
+	get  '/about',	to: 'static_pages#about'
+	get  '/contact',to: 'static_pages#contact'
+	get  '/signup',  to: 'users#new'
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
 end
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: Same comment for this file like as the previous comment and remove the comments.

File: test/controllers/static_pages_controller_test.rb [hunk 28]
```diff
@@ -0,0 +1,30 @@
+require 'test_helper'
```
Comment (mdmahmudurrahman) [inline test/controllers/static_pages_controller_test.rb]: Please remove this auto generated test files, you don't need to care about testing right now.

File: test/controllers/users_controller_test.rb [hunk 29]
```diff
@@ -0,0 +1,9 @@
+require 'test_helper'
```
Comment (mdmahmudurrahman) [inline test/controllers/users_controller_test.rb]: Same comment here like as the previous comment.

File: test/integration/site_layout_test.rb [hunk 30]
```diff
@@ -0,0 +1,12 @@
+require 'test_helper'
```
Comment (mdmahmudurrahman) [inline test/integration/site_layout_test.rb]: Same comment here like as the previous comment.

File: app/assets/stylesheets/custom.scss [hunk 31]
```diff
@@ -0,0 +1,98 @@
+@import "bootstrap-sprockets";
```
Comment (hungnh103) [inline app/assets/stylesheets/custom.scss]: coding convention: use single quote in CSS

**Pull request merged**